File: Programmers/level1/checkScores/check_scores.py
# https://programmers.co.kr/learn/courses/30/lessons/83201#

def solution(scores):
    answer = []

    for i in range(len(scores)):
        tmp = []
        student_score = scores[i][i]  # 자신에게 준 점수
        for j in range(len(scores)):  # 학생의 총 점수를 tmp에
            tmp.append(scores[j][i])
        # print(tmp)
        # print(student_score)
        if max(tmp) == student_score and tmp.count(max(tmp)) == 1:
            tmp.remove(student_score)
        elif min(tmp) == student_score and tmp.count(min(tmp)) == 1:
            tmp.remove(student_score)
        avg = sum(tmp) / len(tmp)
        if avg >= 90:
            answer.append('A')
        elif avg >= 80:
            answer.append('B')
        elif avg >= 70:
            answer.append('C')
        elif avg >= 50:
            answer.append('D')
        else:
            answer.append('F')

    return ''.join(answer)

File: Programmers/level1/checkScores/checkScores.js
// https://programmers.co.kr/learn/courses/30/lessons/83201?language=javascript

export const solution = (scores) => {
    let answer = '';
    for(let i = 0; i < scores.length; i++){
        let tmp = [];
        let student_score = scores[i][i];
        for(let j = 0; j < scores.length; j++){
            tmp.push(scores[j][i]);
        }
        if(Math.max(...tmp) === student_score && tmp.filter(e => e === student_score).length === 1){
            const idx = tmp.indexOf(student_score);
            tmp.splice(idx, 1);
        }
        if(Math.min(...tmp) === student_score && tmp.filter(e => e === student_score).length === 1){
            const idx = tmp.indexOf(student_score);
            tmp.splice(idx, 1);
        }
        // console.log(tmp)
        let avg = tmp.reduce((prev, curr) => (prev + curr)) / tmp.length;
        if(avg >= 90){
            answer += 'A'
        } else if(avg >= 80){
            answer += 'B'
        } else if(avg >= 70){
            answer += 'C'
        } else if(avg >= 50){
            answer += 'D'
        } else {
            answer += 'F'
        }
    }
    return answer;
}